fibnonacci-air/src/lib.rs

```diff
@@ -0,0 +1,136 @@
+use std::borrow::Borrow;
+
+use p3_air::{Air, AirBuilder, AirBuilderWithPublicValues, BaseAir};
+use p3_field::PrimeField64;
+use p3_matrix::dense::RowMajorMatrix;
+use p3_matrix::MatrixRowSlices;
+
+const NUM_FIBONACCI_COLS: usize = 3;
+
+pub struct FibonacciRow<F> {
+    pub index: F,
+    pub left: F,
+    pub right: F,
+}
+
+impl<F> FibonacciRow<F> {
+    fn new(index: F, left: F, right: F) -> FibonacciRow<F> {
+        FibonacciRow {
+            index,
+            left,
+            right,
+        }
+    }
+}
+
+impl<F> Borrow<FibonacciRow<F>> for [F] {
+    fn borrow(&self) -> &FibonacciRow<F> {
+        debug_assert_eq!(self.len(), NUM_FIBONACCI_COLS);
+        let (prefix, shorts, suffix) = unsafe { self.align_to::<FibonacciRow<F>>() };
+        debug_assert!(prefix.is_empty(), "Alignment should match");
+        debug_assert!(suffix.is_empty(), "Alignment should match");
+        debug_assert_eq!(shorts.len(), 1);
+        &shorts[0]
+    }
+}
+
+pub struct FibonacciAir {}
+
+impl<F> BaseAir<F> for FibonacciAir {
+    fn width(&self) -> usize {
+        NUM_FIBONACCI_COLS
+    }
+}
+
+impl<AB: AirBuilderWithPublicValues> Air<AB> for FibonacciAir {
+    fn eval(&self, builder: &mut AB) {
+        let main = builder.main();
+        let pis = builder.public_values();
+
+        let a = pis[0];
+        let b = pis[1];
```

Comment:
I realize this is only a toy example so it shouldn't matter too much, but I'm a bit confused about what `a`, `b` represent. It looks like both must appear on the first row. I would have expected a single public value for the result which must match the last row. I would have expected the first row to be hardcoded to `0, 1`, unless the idea is to allow other starting values?

Reply:
I was going to make it dynamic, proving the n-th item in a fib series is x given the input of a and b. I wanted to write builder.when(1 - (next.index - n)**p).assert_eq(next.b, x), but I didn't find a built in function for exponentiation, so the idea was discarded.

Reply:
I'll just delete this crate and make it a test under uni-stark for the sole purpose of testing public values. How does that sound?

Reply:
i tweaked it to assert the last row is eq to some public value x. I also moved it to uni-stark/tests.